### PR TITLE
Feature/infer check

### PR DESCRIPTION
**Pull request by qweek**
check build with infer

### CODE REVIEW

**Comment by qweek** [comment]
Uber plugin for Infer (#286) is not compatible with new versions
Dependencies installed on Travis in 23 minutes
I think it's better to run Infer in a separate branch

File: .travis.yml [hunk 1]
```diff
@@ -10,12 +10,22 @@ addons:
 
 dist: trusty
 #group: edge
-sudo: false
+sudo: true
 # as per http://blog.travis-ci.com/2014-12-17-faster-builds-with-container-based-infrastructure/
 
+install:
+  - chmod +x gradle/installViaTravis.sh
```
Comment (yschimke) [inline .travis.yml]: shouldn't this be handled when you add the file to git?

File: .travis.yml [hunk 2]
```diff
@@ -10,12 +10,22 @@ addons:
...
 script: gradle/buildViaTravis.sh
 
+after_script:
+  - gradle clean
+  - infer capture -- gradle test
+  - infer analyze -a checkers --print-active-checkers --quiet && cat infer-out/bugs.txt
+  - infer --no-default-checkers --bufferoverrun --compute-analytics --eradicate --repeated-calls --suggest-nullable --tracing --print-active-checkers --quiet && cat infer-out/bugs.txt
+
 # cache between builds
 cache:
```
Comment (yschimke) [inline .travis.yml]: Can you add the infer builds to the cache and check before building them?

File: .travis.yml [hunk 3]
```diff
@@ -10,12 +10,22 @@ addons:
 
 dist: trusty
 #group: edge
-sudo: false
+sudo: true
 # as per http://blog.travis-ci.com/2014-12-17-faster-builds-with-container-based-infrastructure/
```
Comment (yschimke) [inline .travis.yml]: this contradicts the "sudo: true" here.

File: .travis.yml (hunk 2 above)
Comment (qweek) [reply .travis.yml]: Ok, I will check how it works with cache

**Comment by robertroeser** [comment]
@qweek @yschimke what's outstanding for this request? The caching stuff?

**Comment by qweek** [comment]
41 min without cache
5 min 47 sec with cache

**Comment by yschimke** [comment]
I restarted the build - will land if its running in 5 minutes, not 41 :)

**Comment by robertroeser** [comment]
How often is this run? Is it 5 minutes locally? The local build is pretty snappy now, it would nice to keep it that way.

**Comment by yschimke** [comment]
5 minutes would be similar to existing travis builds - https://travis-ci.org/rsocket/rsocket-java/builds, but I haven't seen caching working yet.  I keep seeing 41 minute builds.

This won't run locally without you running extra steps, so shouldn't affect dev experience.

n.b. I love infer, thanks @qweek for adding this, assuming we get it faster than 40 minutes.

**Comment by robertroeser** [comment]
@qweek @yschimke  What's the status of this? Does the caching work now?

**Comment by yschimke** [comment]
I'd love to see this in place, but I didn't see caching work which I think makes it a non starter.

**Pull request closed without merging**